### PR TITLE
fix(datasource/pypi): use pypi.org/pypi registry instead of pypi.org/simple 

### LINKED ISSUE #15938: Missing changelog/release notes for python packages with Pipenv
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Hi,

I started a discussion (#15892) but as I didn't get any solution, I will consider it as a bug.

The problem is that python packages (most of them) does not have any release notes or changelogs when using Pipenv as package manager even though all of the packages are from pypi and publicly accessible.

It seems like Renovate does not manage to extract the necessary URLs. All of the packages lack release notes or even link to the repo, except those which are defined here I guess:
https://github.com/renovatebot/renovate/blob/9f50a7d668fa0e6d56af02005bdb0531691ac025/lib/modules/datasource/metadata-manual.ts#L19-L47

See for example KOliver94/bss-request-manager#785 or KOliver94/bss-request-manager#779.

Minimal reproduction: [KOliver94/renovate-changelog-issue](https://github.com/KOliver94/renovate-changelog-issue)

### Relevant debug logs

<details><summary>Logs</summary>

```
{
  "config": {
    "pipenv": [
      {
        "packageFile": "Pipfile",
        "deps": [
          {
            "depType": "packages",
            "depName": "jsonschema",
            "managerData": {},
            "currentValue": "==4.5.0",
            "datasource": "pypi",
            "depIndex": 0,
            "warnings": [],
            "versioning": "pep440",
            "currentVersion": "4.5.0",
            "isSingleVersion": true,
            "fixedVersion": "4.5.0",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "4.6.0",
                "newValue": "==4.6.0",
                "newMajor": 4,
                "newMinor": 6,
                "updateType": "minor",
                "isRange": true,
                "branchName": "renovate/jsonschema-4.x"
              }
            ]
          },
          {
            "depType": "packages",
            "depName": "sentry-sdk",
            "managerData": {},
            "currentValue": "==1.5.10",
            "datasource": "pypi",
            "depIndex": 1,
            "warnings": [],
            "versioning": "pep440",
            "currentVersion": "1.5.10",
            "isSingleVersion": true,
            "fixedVersion": "1.5.10",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "1.5.12",
                "newValue": "==1.5.12",
                "newMajor": 1,
                "newMinor": 5,
                "updateType": "patch",
                "isRange": true,
                "branchName": "renovate/sentry-sdk-1.x"
              }
            ]
          }
        ],
        "registryUrls": [
          "https://pypi.org/simple"
        ],
        "lockFiles": [
          "Pipfile.lock"
        ],
        "constraints": {
          "python": "== 3.10.*"
        }
      }
    ]
  }
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Replace `https://pypi.org/simple/` with `https://pypi.org/pypi/`. this enables changelog retrieval as the `simple` api doent provide any while the later does.

Wrap `githubRepoPattern` regex to exclude gh sponsors urls such as in this `project_urls` object:
```json

{
  "changelog": "https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst",
  "documentation": "https://python-jsonschema.readthedocs.io/",
  "funding": "https://github.com/sponsors/Julian",
  "homepage": "https://github.com/python-jsonschema/jsonschema",
  "issues": "https://github.com/python-jsonschema/jsonschema/issues/",
  "source": "https://github.com/python-jsonschema/jsonschema",
  "tidelift": "https://tidelift.com/subscription/pkg/pypi-jsonschema?utm_source=pypi-jsonschema&utm_medium=referral&utm_campaign=pypi-link"
}
```

this can also lead to this bug - 
```
DEBUG: Fetching changelog: https://github.com/sponsors/Julian (4.5.0 -> 4.6.0) (repository=ladzaretti/renovate-issue-15938)
DEBUG: Failed to fetch Github tags (repository=ladzaretti/renovate-issue-15938)
       "sourceRepo": "sponsors/Julian",
       "err": {
         "message": "Could not resolve to a Repository with the name 'sponsors/Julian'.",
         "stack": "Error: Could not resolve to a Repository with the name 'sponsors/Julian'.\n    at CacheableGithubTags.getItems (C:\\projects\\ladzaretti-renovate\\lib\\modules\\
datasource\\github-releases\\cache\\cache-base.ts:259:19)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async getTags (C:\\projects\\ladzaretti
-renovate\\lib\\workers\\repository\\update\\pr\\changelog\\github\\index.ts:31:18)\n    at async getRef (C:\\projects\\ladzaretti-renovate\\lib\\workers\\repository\\update\\pr\\c
hangelog\\source-github.ts:109:14)\n    at async Object.getChangeLogJSON (C:\\projects\\ladzaretti-renovate\\lib\\workers\\repository\\update\\pr\\changelog\\source-github.ts:149:2
6)\n    at async getChangeLogJSON (C:\\projects\\ladzaretti-renovate\\lib\\workers\\repository\\update\\pr\\changelog\\index.ts:36:15)\n    at async embedChangelog (C:\\projects\\l
adzaretti-renovate\\lib\\workers\\repository\\changelog\\index.ts:7:21)\n    at async C:\\projects\\ladzaretti-renovate\\node_modules\\p-map\\index.js:57:22"
       }
DEBUG: GitHub 404 (repository=ladzaretti/renovate-issue-15938)
       "url": "https://api.github.com/repos/sponsors/Julian"
DEBUG: Error 404 getting changelog md (repository=sponsors/Julian)
       "type": "github",
       "apiBaseUrl": "https://api.github.com/"

```

## Context
Closes #15938

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by Gabriel-Ladzaretti** [comment]
```
DEBUG: Fetching changelog: https://github.com/python-jsonschema/jsonschema (4.5.0 -> 4.6.0) (repository=ladzaretti/renovate-issue-15938)
DEBUG: Fetching changelog: https://github.com/getsentry/sentry-python (1.5.10 -> 1.6.0) (repository=ladzaretti/renovate-issue-15938)
DEBUG: config.repoIsOnboarded=true (repository=ladzaretti/renovate-issue-15938)
DEBUG: packageFiles with updates (repository=ladzaretti/renovate-issue-15938)
       "config": {
         "pipenv": [
           {
             "packageFile": "Pipfile",
             "deps": [
               {
                 "depType": "packages",
                 "depName": "jsonschema",
                 "managerData": {},
                 "currentValue": "==4.5.0",
                 "datasource": "pypi",
                 "depIndex": 0,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "4.6.0",
                     "newValue": "==4.6.0",
                     "releaseTimestamp": "2022-06-01T21:16:22.000Z",
                     "newMajor": 4,
                     "newMinor": 6,
                     "updateType": "minor",
                     "isRange": true,
                     "branchName": "renovate/jsonschema-4.x"
                   }
                 ],
                 "warnings": [],
                 "versioning": "pep440",
                 "sourceUrl": "https://github.com/python-jsonschema/jsonschema",
                 "changelogUrl": "https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst",
                 "currentVersion": "4.5.0",
                 "isSingleVersion": true,
                 "fixedVersion": "4.5.0"
               },
               {
                 "depType": "packages",
                 "depName": "sentry-sdk",
                 "managerData": {},
                 "currentValue": "==1.5.10",
                 "datasource": "pypi",
                 "depIndex": 1,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "1.6.0",
                     "newValue": "==1.6.0",
                     "releaseTimestamp": "2022-06-22T08:27:10.000Z",
                     "newMajor": 1,
                     "newMinor": 6,
                     "updateType": "minor",
                     "isRange": true,
                     "branchName": "renovate/sentry-sdk-1.x"
                   }
                 ],
                 "warnings": [],
                 "versioning": "pep440",
                 "sourceUrl": "https://github.com/getsentry/sentry-python",
                 "changelogUrl": "https://github.com/getsentry/sentry-python/blob/master/CHANGELOG.md",
                 "currentVersion": "1.5.10",
                 "isSingleVersion": true,
                 "fixedVersion": "1.5.10"
               }
             ],
             "registryUrls": ["https://pypi.org/simple"],
             "constraints": {"python": "== 3.10.*"}
           }
         ]
       }

```

File: lib/modules/datasource/pypi/index.spec.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { PypiDatasource } from '.';
 
 const res1 = Fixtures.get('azure-cli-monitor.json');
 const res2 = Fixtures.get('azure-cli-monitor-updated.json');
+const res3 = Fixtures.get('six.json');
```
Comment (viceice) [inline lib/modules/datasource/pypi/index.spec.ts:9]: do we really need a new fixture?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/datasource/pypi/index.ts [hunk 2]
```diff
@@ -35,7 +34,9 @@ export class PypiDatasource extends Datasource {
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
     let dependency: ReleaseResult | null = null;
-    const hostUrl = ensureTrailingSlash(`${registryUrl}`);
+    const hostUrl = ensureTrailingSlash(
```
Comment (viceice) [inline lib/modules/datasource/pypi/index.ts:37]:
```suggestion
    // TODO: null check (#7154)
    const hostUrl = ensureTrailingSlash(
```

**Review by viceice — APPROVED** [review]

**Pull request merged**